gtfu/src/web/graas-report-web.js

```diff
@@ -97,26 +118,57 @@ function handleAgencyChoice(){
 
 function handleDateChoice(){
     console.log("handleDateChoice()");
+    isNewPageSelect = true;
     selectedDate = document.getElementById("date-select").value;
-    url = `${bucketURL}/graas-report-archive/${selectedAgency}/${selectedAgency}-${selectedDate}.json`;
+    // "?nocache=" mostly forces the page to reload files each time
+    url = `${bucketURL}/graas-report-archive/${selectedAgency}/${selectedAgency}-${selectedDate}.json?nocache=${(new Date()).getTime()}`;
     loadJSON(url, processTripJSON);
 }
 
 function processTripJSON(object){
     trips.length = 0;
-    for (var i = 0; i < object.trips.length; i++) {
-        var map = {x: object["trips"][i]["boundaries"]["map-x"],
+    for (let i = 0; i < object.trips.length; i++) {
+        let map = {x: object["trips"][i]["boundaries"]["map-x"],
                    y: object["trips"][i]["boundaries"]["map-y"],
                    width: object["trips"][i]["boundaries"]["map-width"],
                    height: object["trips"][i]["boundaries"]["map-height"]
                 };
-        var timeline = {x: object["trips"][i]["boundaries"]["timeline-x"],
+        let timeline = {x: object["trips"][i]["boundaries"]["timeline-x"],
                        y: object["trips"][i]["boundaries"]["timeline-y"],
                        width: object["trips"][i]["boundaries"]["timeline-width"],
                        height: object["trips"][i]["boundaries"]["timeline-height"]
                 };
+        let points = []
+        for (let j = 0; j < object["trips"][i]["trip-points"].length; j++) {
+            points.push({x: object["trips"][i]["trip-points"][j]["x"],
+                        y: object["trips"][i]["trip-points"][j]["y"],
+                        secs: Math.round(object["trips"][i]["trip-points"][j]["millis"]/1000),
+                        count: object["trips"][i]["trip-points"][j]["count"]
+                        });
+        }
+        // Sort points
+        points.sort(function compareFn(a, b) {
+            if (a.secs <= b.secs) {
+                return -1;
+            }
+            else return 1;
+        });
+
+        // Remove points with duplicate timestamps - this is rare but it happens.
+        // A more robust approach would be to prevent duplicate timestamps from occuring.
```

Comment:
I don't understand what's happening here. First time through x is 0 and y is 1. If the timestamps for the respective elements in the point list are different, increase x, making it 1 also. Then copy points[y] into points[x], which is a nop since x and y are equal. If timestamps are the same, just increase y, so now x is 0 and y is 2?
Are you sure this does what you think it does? Please provide more detailed comment.

Reply:
I do believe it does what I think. It  I should have mentioned in a comment that I modified the logic from a [blog post](https://medium.com/swlh/javascript-remove-duplicates-from-sorted-array-in-place-d3d959fb4d77). 

It basically updates the array in place so that the first N points have unique timestamps. Then I splice it it to cut off the remaining points. I will update comments to be a bit more descriptive and link the source.